pkg/codegen/internal/test/testdata/cyclic-types/go/example/provider.go

```diff
@@ -48,7 +48,7 @@ type ProviderInput interface {
 }
 
 func (*Provider) ElementType() reflect.Type {
-	return reflect.TypeOf((*Provider)(nil))
+	return reflect.TypeOf((**Provider)(nil)).Elem()
```

Comment:
This is a no-op change right? It seems to me before and after is the same. Would it be possible to special-case the generator a bit to avoid the change. It's a minor thing here but it's really annoying in provider upgrades to be seeing  more diffs than strictly needed for the change.

Reply:
This is a no-op change, yes. 

> Would it be possible to special-case the generator a bit to avoid the change.

It would, though that sort of special-casing is what I was hoping to avoid with these changes (I removed the special-case code that produced the original output).